lib/dependencies/RequireEnsureDependenciesBlockParserPlugin.js

```diff
@@ -52,18 +73,24 @@ module.exports = class RequireEnsureDependenciesBlockParserPlugin {
 							if(failed) {
 								return;
 							}
-							if(fnExpression) {
-								if(fnExpression.fn.body.type === "BlockStatement")
-									parser.walkStatement(fnExpression.fn.body);
+							if(successExpression) {
+								if(successExpression.fn.body.type === "BlockStatement")
+									parser.walkStatement(successExpression.fn.body);
+								else
+									parser.walkExpression(successExpression.fn.body);
+							}
+							if(errorExpression) {
```

Comment:
This part of the code should be moved after the finally block. If it's here dependencies in this block are put into the chunk. This doesn't make sense for loading errors. If it's after the finally block, dependencies are not put into the chunk.